Day-1/Theory/README.md

```diff
@@ -0,0 +1,244 @@
+# Day 1 - Theory
+
+## Table of contents
+
+- [Introduction and Architecture](#introduction-and-architecture)
+- [JSX](#jsx)
+- [Virtual DOM](#virtual-dom)
+- [Rendering Elements](#rendering-elements)
+- [Handling Events](#handling-events)
+- [Routes and Navigation](#routes-and-navigation)
+
+## Introduction and Architecture
+
+- [React](https://reactjs.org/) is a JavaScript library for building interfaces
+- the components manage their own state
+- all components take input data and return what to display
+- we can easily write our code in the component using [JSX](https://reactjs.org/docs/introducing-jsx.html), an XML-like syntax extension for JavaScript
+- [TypeScript](https://www.typescriptlang.org/) is a statically compiled language to write clear and simple JavaScript code
+- TypeScript is converted into plain JavaScript code
+- the main benefits of using TypeScript are:
+  - instant type error detection
+  - better IDE experience
+  - more readable code
+- TypeScript can be used with JSX: [TypeScript for JSX](https://www.typescriptlang.org/docs/handbook/jsx.html)
+- here we have an example with a very simple React component:
+
+    ```JavaScript
+    class Hello extends React.Component {
+        render() {
+            return (
+                <div>Hello {this.props.name}</div>
+            );
+        }
+    }
+
+    ReactDOM.render(
+        <Hello name="World" />,
+        document.getElementById('our-first-example');
+    )
+    ```
+
```

Comment:
We need to speak more about how to split UI into components

Reply:
I added a new section - Why and how to split UI into components. Please review. Thanks!